dist/Makefile.tmpl

```diff
@@ -49,6 +49,9 @@ include Makefile.basic
 
 CFLAGS		+= -Wno-parentheses -Wno-deprecated-declarations -Wno-\#warnings -Wno-error=cpp -Wno-cpp -g -std=gnu11 -O3
 
+CFLAGS_AES_NI = -maes -msse4.1
```

Comment:
probably doesn't work for ARM at all -- you probably need to add further changes into `dist/configure` and do something like

```
compile_aesni=true
```

(in `detect_x64`)

then 

```
if ! $compile_aesni; then
  echo "$CC $CROSS_CFLAGS cannot compile NI, disabling AES-NI"
  echo "BLACKLIST += $(ls *AES*NI*.c | xargs)" >> Makefile.config
else
  echo "#define HACL_CAN_COMPILE_AESNI 1" >> config.h
fi
```

Reply:
and running your test would then be guarded in tests/Makefile just like, say, the vec128 test, for instance

Reply:
I made this change, however I am not able test this function https://github.com/mamonet/hacl-star/blob/aes-ctr32/dist/configure#L203 I would like for you to review the change and test this function to make sure everything's ok

Reply:
on aarch64 (raspberry pi linux):

```
jonathan@raspberrypi:/tmp/hacl-star/dist/gcc-compatible $ ./configure
bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
... detected ARM platform
...  supports 128-bit NEON
... cc  can cross-compile to ARM64 with SIMD
...  supports cryptographic instructions
... cc  can cross-compile to ARM64 with crypto extension
 does not support x64 assembly, disabling Curve64
 does not support legacy vale stubs
 does not support 256-bit arithmetic
... glibc is recent enough for explicit_bzero
```

on apple m3 (osx)

```
... detected ARM platform
...  supports 128-bit NEON
... cc  can cross-compile to ARM64 with SIMD
...  supports cryptographic instructions
... cc  can cross-compile to ARM64 with crypto extension
 does not support x64 assembly, disabling Curve64
 does not support legacy vale stubs
 does not support 256-bit arithmetic
 ```

that doesn't mean the raspberry pi can *run* aesni but at least it can compile it

so this part looks good to me